go/internal/controller/mcp_server_controller.go

```diff
@@ -51,11 +54,16 @@ func (r *MCPServerController) Reconcile(ctx context.Context, req ctrl.Request) (
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *MCPServerController) SetupWithManager(mgr ctrl.Manager) error {
+	if _, err := mgr.GetRESTMapper().RESTMapping(mcpServerGK); err != nil {
+		ctrl.Log.Info("MCPServer CRD not found - controller will not be started", "controller", "mcpserver")
+		return nil
+	}
+
 	return ctrl.NewControllerManagedBy(mgr).
 		WithOptions(controller.Options{
 			NeedLeaderElection: ptr.To(true),
 		}).
 		For(&v1alpha1.MCPServer{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
-		Named("toolserver").
+		Named("mcpserver").
```

Comment:
This change wasn't possible previously (see https://github.com/kagent-dev/kagent/pull/757) because the code was already running [a controller named `mcpserver`](https://github.com/kagent-dev/kmcp/blob/41071d12350f22dd20492f5848e0629ba5ee7c0c/pkg/controller/mcpserver_controller.go#L108). I.e. this code was running 2 separate controllers reconciling the same resources. :anguished: